File: src/com/peyo/rplauncher/GridFragment.java
package com.peyo.rplauncher;

import java.util.ArrayList;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v17.leanback.app.VerticalGridFragment;
import android.support.v17.leanback.widget.ArrayObjectAdapter;
import android.support.v17.leanback.widget.OnItemViewClickedListener;
import android.support.v17.leanback.widget.Presenter.ViewHolder;
import android.support.v17.leanback.widget.RowPresenter;
import android.support.v17.leanback.widget.Row;
import android.support.v17.leanback.widget.VerticalGridPresenter;

public class GridFragment extends VerticalGridFragment {
    private static final int NUM_COLUMNS = 5;
    private ArrayObjectAdapter mAdapter = new ArrayObjectAdapter(
            new IconPresenter());

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setAdapter(mAdapter);
        VerticalGridPresenter gridPresenter = new VerticalGridPresenter();
        gridPresenter.setNumberOfColumns(NUM_COLUMNS);
        setGridPresenter(gridPresenter);
        setOnItemViewClickedListener(new OnItemViewClickedListener() {
            @Override
            public void onItemClicked(ViewHolder viewHolder, Object item,
                                      RowPresenter.ViewHolder rowViewHolder, Row row) {
                AppInfo info = (AppInfo) item;
                Intent intent = new Intent();
                intent.setComponent(info.getComponentName());
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
                viewHolder.view.getContext().startActivity(intent);
            }
        });
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        getView().setBackgroundColor(Color.BLACK);
    }

    public void updateApps(ArrayList<AppInfo> apps) {
        for (int i = 0; i < apps.size(); i++) {
            mAdapter.add(apps.get(i));
        }
    }
}

File: src/com/arpi/rplauncher/GridFragment.kt
package com.arpi.rplauncher

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import androidx.leanback.app.VerticalGridFragment
import androidx.leanback.widget.ArrayObjectAdapter
import androidx.leanback.widget.OnItemViewClickedListener
import androidx.leanback.widget.VerticalGridPresenter
import java.util.ArrayList


class GridFragment : VerticalGridFragment() {
    private val mAdapter = ArrayObjectAdapter(
            IconPresenter())

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        adapter = mAdapter
        val gridPresenter = VerticalGridPresenter()
        gridPresenter.numberOfColumns = NUM_COLUMNS
        setGridPresenter(gridPresenter)
        onItemViewClickedListener = OnItemViewClickedListener { viewHolder, item, _, _ ->
            val info = item as AppInfo
            val intent = Intent()
            intent.component = info.componentName
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED)
            viewHolder.view.context.startActivity(intent)
        }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        view!!.setBackgroundColor(Color.BLACK)
    }

    fun updateApps(apps: ArrayList<AppInfo>) {
        for (i in apps.indices) {
            mAdapter.add(apps[i])
        }
    }

    companion object {
        private val NUM_COLUMNS = 5
    }
}
